src/joint.dia.link.js

```diff
@@ -1773,6 +1795,18 @@ joint.dia.LinkView = joint.dia.CellView.extend({
                 this._z = null;
             }
 
+            var currentEnd = this.model.prop(arrowhead) || {};
+            var changed = initialEnd.id !== currentEnd.id || initialEnd.port !== currentEnd.port;
```

Comment:
This is not safe enough. e.g. a change is detected for `{ id: 'a', port: null }` ==> `{ id: 'a' }`.